app/controllers/application_controller.rb

```diff
@@ -20,9 +20,57 @@ class ApplicationController < ActionController::Base
 
   def auth_user!
     unless user_signed_in? || devise_controller?
-      session[:previous_url] = request.fullpath unless request.xhr? # do not store AJAX calls
+      session[:previous_url] = request.fullpath unless request.xhr? # do not store AJAX call
       redirect_to new_user_session_path
     end
+    check_path
+  end
+
+  def find_room(id)
+    rooms = current_user.rooms.eager_load(:learning_objects)
+    lo = LearningObject.find(id)
+    room =  rooms.reject{ |m| !(m.learning_objects.include?(lo)) }.first
+    if room.present?
+      params[:room_number] = room.id
+      params[:week_number] = room.week_id
+    end
+  end
+
+  def check_path
+    if current_user.present?
```

Comment:
Tato metoda je priserne necitatelna.
- Aby to nebolo povnarane na n-tej urovni a cele uzavrete v dvoch velkych ifoch, tak je lepsie na zaciatku vylucit pripad, ze to neplati, v zmysle `return unless current_user.andand.has_rooms?`
- Vo vsetkych tych podmienkach sa stale dookola opakuje, ze bud musi existovat `params[:room_number]` alebo `params[:week_number]`. Potom by bolo dobre to rozdelit na najvyssej urovni na tie tri pripady (je vyplnene jedno, je vyplnene druhe alebo su vyplnene obe) a potom uz v ramci tychto troch casti riesit jednotlive pripady, ci tam je id, nie je id, atd.)
- netreba zbytocne pouzivat vsade `.present?`, present je `!blank?`, co rozlisuje prazdne stringy, prazdne polia, prazdne hashe, ak nam ide iba o to overit, ci nieco nie je nil (takze technicky `!nil?`), staci pouzit samotnu premennu napr. `if current_user`. A ak sa ten if robi iba pre to, aby sa na to nasledne mohla zavolat metoda, tak sa da pouzit gem `andand`, takze z
```
if x.present?
  if x.has_rooms?
```
sa stane jednoducho `if x.andand.has_rooms?`